extensions/common/sql/common-sql/src/main/java/org/eclipse/dataspaceconnector/sql/SqlDefaultServicesExtension.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql;
+
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Extension;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Provider;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.datasource.DataSourceRegistry;
+import org.eclipse.dataspaceconnector.sql.datasource.DefaultDataSourceRegistry;
+
+/**
+ * Default SQL services. It provides a default {@link DataSourceRegistry} without a transaction manager
+ */
+@Extension(value = SqlDefaultServicesExtension.NAME)
+public class SqlDefaultServicesExtension implements ServiceExtension {
+
+    public static final String NAME = "SQL Default Services";
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+
+    @Provider(isDefault = true)
+    public DataSourceRegistry dataSourceRegistry(ServiceExtensionContext context) {
```

Comment:
Could we move this to `CoreDefaultServicesExtension`? That would introduce a dependency onto `transaction-datasource-spi` in `core:common:base`, but to me that's OK, since the `NoopTransactionContext` (and thus a dependency onto `transaction-spi`) is already registered there.